icicle/include/api/babybear.h

```diff
@@ -32,47 +32,49 @@ extern "C" cudaError_t babybear_create_poseidon2_cuda(
   device_context::DeviceContext& ctx
 );
 
-extern "C" cudaError_t babybear_init_poseidon2_cuda(
+extern "C" cudaError_t babybear_poseidon2_load_cuda(
   poseidon2::Poseidon2<babybear::scalar_t>** poseidon,
   unsigned int width,
   poseidon2::MdsType mds_type,
   poseidon2::DiffusionStrategy diffusion,
   device_context::DeviceContext& ctx
 );
 
-extern "C" cudaError_t babybear_poseidon2_permute_many_cuda(
+extern "C" cudaError_t babybear_poseidon2_absorb_many_cuda(
+  const poseidon2::Poseidon2<babybear::scalar_t>* poseidon,
+  const babybear::scalar_t* inputs,
+  babybear::scalar_t* states,
+  unsigned int number_of_states,
+  unsigned int input_block_len,
+  hash::SpongeConfig& cfg);
+
+extern "C" cudaError_t babybear_poseidon2_squeeze_many_cuda(
   const poseidon2::Poseidon2<babybear::scalar_t>* poseidon,
   const babybear::scalar_t* states,
   babybear::scalar_t* output,
   unsigned int number_of_states,
-  device_context::DeviceContext& ctx,
-  bool is_async
-);
+  unsigned int output_len,
+  hash::SpongeConfig& cfg);
 
-extern "C" cudaError_t babybear_poseidon2_compress_many_cuda(
+extern "C" cudaError_t babybear_poseidon2_hash_many_cuda(
   const poseidon2::Poseidon2<babybear::scalar_t>* poseidon,
-  const babybear::scalar_t* states,
+  const babybear::scalar_t* inputs,
   babybear::scalar_t* output,
   unsigned int number_of_states,
-  unsigned int offset,
-  babybear::scalar_t* perm_output,
-  device_context::DeviceContext& ctx,
-  bool is_async
-);
+  unsigned int input_block_len,
+  unsigned int output_len,
+  hash::SpongeConfig& cfg);
 
-extern "C" cudaError_t babybear_release_poseidon2_constants_cuda(
-  poseidon2::Poseidon2Constants<babybear::scalar_t>* constants,
-  device_context::DeviceContext& ctx);
+extern "C" cudaError_t
+  babybear_poseidon2_delete_cuda(poseidon2::Poseidon2<babybear::scalar_t>* poseidon, device_context::DeviceContext& ctx);
 
-extern "C" cudaError_t babybear_build_poseidon_merkle_tree(
+extern "C" cudaError_t babybear_build_poseidon2_merkle_tree(
   const babybear::scalar_t* leaves,
   babybear::scalar_t* digests,
   unsigned int height,
-  unsigned int arity,
```

Comment:
why did you remove arity? is it in the config now?

Reply:
exactly. Actually, both variants are fine I guess, but given the fact that most of the trees are binary - adding it to config seems better